### PR TITLE
main code cleanup

### PR DESCRIPTION
**Pull request by tovernaar123**
many things changed

### CODE REVIEW

File: commands/playersonline-new.js [hunk 1]
```diff
@@ -1,366 +1,67 @@
...
+    await msg.channel.send("Asked for all online players: Awaiting reply from servers...")
+    let results = [];
+    for (let i = 1; i < 9; i++) {
+        let res = oneCommand(i).catch((err)=>{internal_error(err); return})
+        results.push(res)
+    }
+    let values = await Promise.all(results)
+    for (let i = 0; i < values.length; i++) {
+        if(!values[i]){
+            await msg.channel.send(`s${i+1} did not responde`);
```
Comment (aldldl) [inline commands/playersonline-new.js]:
```suggestion
            await msg.channel.send(`s${i+1} did not respond`); 
```

File: commands/playersonline-new.js [hunk 2]
```diff
@@ -1,366 +1,67 @@
...
-            port: `${rconport}`,
-            password: `${rconpw}`
-        });
-
-
+        const server =  Number(args[0]) || args[0];
 
         if (!server) { // Checks to see if the person specified a server number, then checks to see if the server number is part of the array of the servers it could be (1-8 currently)
-            msg.channel.send('Please pick a server first just a number (1-8). \`<#> <username> <reason>\`');
+            msg.channel.send('Please pick a server first just a number (1-8). \`<#> <username> <reason>\`')
```
Comment (aldldl) [inline commands/playersonline-new.js]:
```suggestion
            msg.channel.send(`Please pick a server first. Just the number (1-8) or *all*.  Correct usage is \` po <server#>\``)
```

File: commands/playersonline-new.js [hunk 3]
```diff
@@ -1,366 +1,67 @@
...
+                .catch((err)=>{internal_error(err); return})
             console.log(`Kick-Did not have server number`);
```
Comment (aldldl) [inline commands/playersonline-new.js]:
```suggestion
            console.log(`Players online - No server number provided`);
```

File: commands/playersonline-new.js [hunk 4]
```diff
@@ -1,366 +1,67 @@
...
             return;
         }
-        if (snum.indexOf(server) > -1) {
+
+        if (server >= 1 && server <= 1) {
```
Comment (aldldl) [inline commands/playersonline-new.js]: Maybe I'm missing this but I thought we fixed it earlier - should be > 1 <= 8 right?

```suggestion
        if (server >= 1 && server <= 8) {
```

File: commands/playersonline-new.js [hunk 5]
```diff
@@ -1,366 +1,67 @@
...
+                        .catch((err)=>{internal_error(err); return})
+                })
+                .catch((err)=>{internal_error(err); return})
+        } else if (server === 'all') {
             console.log(`Server is all`);
+            allCommand(msg, internal_error).catch((err)=>{internal_error(err); return})
         } else {
             // If a person DID give a server number but did NOT give the correct one it will return without running - is the server number is part of the array of the servers it could be (1-8 currently)
-            msg.channel.send(`Please pick a server first just a number (1-8) or *all*.  Correct usage is \` po <server#>\``);
+            msg.reply(`Please pick a server first just a number (1-8) or *all*.  Correct usage is \` po <server#>\``)
```
Comment (aldldl) [inline commands/playersonline-new.js]:
```suggestion
            msg.reply(`Please pick a server first. Just the number (1-8) or *all*.  Correct usage is \` po <server#>\``)
```

File: botinfo.js [hunk 6]
```diff
@@ -39,15 +40,20 @@ let seconds = date_ob.getSeconds();
...
 client.on("ready", () => {
     console.log(year + "-" + month + date + " " + hours + ":" + minutes + ":" + seconds + ": I am ready!");
 });
 
+
+
 client.on("message", msg => {
+    function internal_error(err) {
+        console.error(err)
+        msg.channel.send('Internal error in the command plz contact and admin')
```
Comment (aldldl) [inline botinfo.js]:
```suggestion
        msg.channel.send('Internal error in the command. Please contact an admin')
```

**Review by aldldl — CHANGES_REQUESTED** [review]

File: commands/adminsonline.js [hunk 7]
```diff
@@ -1,75 +1,108 @@
...
+        //add and empty to make it look nice 
+        Embed.addField(`\u200B`, `\u200B`, true)
+    }
+    await msg.channel.send(Embed)
+}
+
 module.exports = {
     name: 'ao',
-    aka: ['adminonline','adminsonline','ao'],
+    aka: ['adminonline', 'adminsonline'],
```
Comment (aldldl) [inline commands/adminsonline.js:67]:
```suggestion
    aka: ['adminonline', 'adminsonline', 'admins'],
```

File: commands/adminsonline.js [hunk 8]
```diff
@@ -1,75 +1,108 @@
...
+        let server = Number(args[0]) || args[0]
+        const extra = args[1]; // nothing extra please for this command
 
-        if (!server) 
-        { // Checks to see if the person specified a server number, then checks to see if the server number is part of the array of the servers it could be (1-8 currently)
-            msg.channel.send(`Please pick a server. Just the number - ie S1 would be \`1\` (Currently 1-8). Correct usage is \`ao <Server#>\``);
-            console.log(`AO Check does not have server number`);
-            return;
+        if(!isNaN(server)){
+            server = Math.floor(args[0])
```
Comment (aldldl) [inline commands/adminsonline.js:78]: Can you explain what this means?

File: commands/adminsonline.js [hunk 9]
```diff
@@ -1,75 +1,108 @@
...
         }
-        if (server < 9 && server > 0) 
-        { // If a person DID give a server number but did NOT give the correct one it will return without running - is the server number is part of the array of the servers it could be (1-8 currently)
-            msg.channel.send(`This does not seem to be a valid server number. We need just the number (1-8) ie S1 would be \`1\`.\nCorrect usage is \`ao <Server#>\``);
-            console.log(`AO provided an invalid server number`);
+
+        if (!server) { // Checks to see if the person specified a server number, then checks to see if the server number is part of the array of the servers it could be (1-8 currently)
```
Comment (aldldl) [inline commands/adminsonline.js:81]: This does not actually check the number anymore, it checks that there is not a blank.

File: commands/jail.js [hunk 10]
```diff
@@ -1,67 +1,63 @@
...
-
-
-
-
-            msg.channel.send(responses + ` If no error above user **${tojail}** should have been jailed for *${reason1}*, check with someone to be sure because ALo wrote this...`);
-            console.log(responses + `.`);
-
-            rcon.end()
+        if (server < 9 && server > 0) {
+            console.log('Server is 1-8');
```
Comment (aldldl) [inline commands/jail.js:53]:
```suggestion
            console.log('Server is ${server}');
```

File: commands/kick.js [hunk 11]
```diff
@@ -1,82 +1,64 @@
...
+    await rcon.send(`/kick ${toKick} ${reason}`);
+    await msg.channel.send(`User **${toKick}** has been kicked for *${reason}*, check with someone on S${server} to be sure `);
+    const Embed = Discord.MessageEmbed()
+    Embed.addField('Kick', `A player has been kicked`, false);
+    Embed.addField(`Server Details`, `server: S${server}`, false);
+    Embed.addField(`Player`, `${toKick}`, true);
+    Embed.addField(`By`, `${msg.author.username}`, true);
+    Embed.addField(`Reason`, `${reason}`, true);
+    Embed.setColor("0xb40e0e");
+    let reportChan = msg.guild.channels.cache.get('764881627893334047'); // Reports channel is "368812365594230788" for exp
```
Comment (aldldl) [inline commands/kick.js:17]:
```suggestion
    let reportChan = msg.guild.channels.cache.get('764881627893334047'); // Reports channel is "368812365594230788" for exp // Reports Channel is "764881627893334047" for test server
```

File: commands/log.js [hunk 12]
```diff
@@ -1,63 +1,54 @@
...
+    readLastLines.read(`/home/factorio/servers/eu-0${server}/console.log`, size) //home/factorio/servers/eu-0${server}/console.log
+        .then((lines) => msg.channel.send(lines, { split: true })) // lines gets returned with last X (see above) lines and sent to same channel as command
+        .then(console.log(`The last ${size} lines of the log for server #${server} posted in ${msg.channel.name}. (asked for ${size})`)) // logs the command in the console log
+        .catch((err) => { internal_error(err); return });
+}
+
 module.exports = {
     name: 'log',
     aka: ['dlchat', 'logs'],
     description: 'get chat (last 10 lines) (Admin/Mod only command)',
```
Comment (aldldl) [inline commands/log.js:12]:
```suggestion
    description: 'get chat (last 10 lines) (Board + only command)',
```

File: commands/log.js [hunk 13]
```diff
@@ -1,63 +1,54 @@
...
+    execute(msg, args, _, internal_error) {
+        const server = Math.floor(Number(args[0]));
         let size = args[1];
-        let snum = [`1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`];
         let sizeLimit = 50;
         let defaultSize = 10;
 
         if (!server) {
-            msg.channel.send('Please pick a server first just a number (1-8)');
+            msg.channel.send('Please pick a server first just a number (1-8)')
```
Comment (aldldl) [inline commands/log.js:25]:
```suggestion
            msg.channel.send('Please pick a server first. Just the number (1-8)')
```

File: commands/unjail.js [hunk 14]
```diff
@@ -1,72 +1,65 @@
...
+const Discord = require('discord.js');
+/**
+ * 
+ * @param {Number} server 
+ * @param {Rcon} rcon 
+ * @param {Discord.Message} msg 
+ * @param {string} reason 
+ * @param {string} tojail
+*/
+async function runcommand(server, rcon, msg, to_un_jail) {
```
Comment (aldldl) [inline commands/unjail.js:10]: why to_un_jail when jail is toJail? kick is toKick

File: commands/unjail.js [hunk 15]
```diff
@@ -1,72 +1,65 @@
...
+    if (!rcon.connected) {
+        await msg.channel.send(`S${server} is not connected the bot.`)
+        return;
+    }
+    let res = await rcon.send(`/unjail ${to_un_jail}`)
+    if (res === "Command Complete\n") {
+        await msg.channel.send(`**${to_un_jail}** has been jailed on S${server} for.`);
```
Comment (aldldl) [inline commands/unjail.js:17]:
```suggestion
        await msg.channel.send(`**${to_un_jail}** has been unjailed on S${server}`);
```

File: commands/unjail.js [hunk 16]
```diff
@@ -1,72 +1,65 @@
...
+        const Embed = Discord.MessageEmbed()
+        Embed.addField('UnJail', `A player has been Unjailed`, false);
+        Embed.addField(`Server Details`, `server: S${server}`, false);
+        Embed.addField(`Player`, `${to_un_jail}`, true);
+        Embed.addField(`By`, `${msg.author.username}`, true);
+        Embed.setColor("0xb40e0e");
+        let reportChan = msg.guild.channels.cache.get('764881627893334047');
+        await reportChan.send(Embed);
+        console.log(`${msg.author.username} has jailed ${to_un_jail} on S${server} for`);
```
Comment (aldldl) [inline commands/unjail.js:26]:
```suggestion
        console.log(`${msg.author.username} has unjailed ${to_un_jail} on S${server}.`);
```

File: commands/unjail.js [hunk 17]
```diff
@@ -1,72 +1,65 @@
...
             return;
         }
-        //if(!reason1){
-        //	msg.channel.send(`Please pick a server first just a number (1-8).  Correct usage is jail \`<#> <username> <reason>\``);
-        //	return;
-        //}
-        if (!tojail) {
-            msg.channel.send(`You need to tell us who you would like to jail for us to be able to jail them`);
+        if (server < 9 && server > 0) {
+            console.log('Server is 1-8');
```
Comment (aldldl) [inline commands/unjail.js:55]:
```suggestion
            console.log('Unjail server is S${server}');
```

File: commands/unjail.js [hunk 18]
```diff
@@ -1,72 +1,65 @@
...
+            runcommand(server, rcons[server], msg, to_un_jail)
+                .catch((err) => { internal_error(err); return })
+        } else {
+            msg.reply(`Server number can\'t be bigger then 8 or smaller then 1. Correct usage is \` .exp unjail <server#> <username>\``)
```
Comment (aldldl) [inline commands/unjail.js:59]:
```suggestion
            msg.reply(`Please pick a valid server first. Just the number (currently 1-8). Correct usage is \` unjail <server#> <username>\``)
```

File: commands/unjail.js [hunk 19]
```diff
@@ -1,72 +1,65 @@
...
+                .catch((err) => { internal_error(err); return })
+            console.log(`jail by ${msg.author.username} incorrect server number`);
```
Comment (aldldl) [inline commands/unjail.js:61]:
```suggestion
            console.log(`unjail by ${msg.author.username} used the incorrect server number`);
```

File: infoBot.js [hunk 20]
```diff
@@ -0,0 +1,151 @@
...
+});
+
+
+
+
+client.on("message", async msg => {
+    
+    function internal_error(err) {
+        console.log(err)
+        msg.channel.send('Internal error in the command plz contact and admin')
```
Comment (aldldl) [inline infoBot.js:92]:
```suggestion
        msg.channel.send('Internal error in the command. Please contact an admin.')
```

File: infoBot.js [hunk 21]
```diff
@@ -0,0 +1,151 @@
...
+role = {
+    staff: "482924291084779532",
+    admin: "290940523844468738",
+    mod: "260893080968888321",
+    board: "478962844604301312"
+}
+//
+
+//array for all ofline servers
+let offline_servers = [2, 7]
```
Comment (aldldl) [inline infoBot.js:32]: Still not a fan of keeping 2 & 7 never connecting.

File: rcon_auto_connect.js [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+        });
+    }
+
+    client.on("end", function() {
+        //Reconnect if a successfull connection was made.
+        if (connected) {
+            console.log(`Lost connection with S${i}`);
+            console.log("Reconnecting in 10 seconds");
+            connected = false;
+            setTimeout(connect, 10e3).unref();
```
Comment (aldldl) [inline rcon_auto_connect.js:52]: Can we make longer, this spams the logs and seems too often, maybe 30 sec or 1 min?

**Review by aldldl — CHANGES_REQUESTED** [review]

**Comment by aldldl** [comment]
Not sure if you saw but I did a second code review shortly after the first one for the other files

File: commands/adminsonline.js (hunk 8 above)
Comment (tovernaar123) [reply commands/adminsonline.js:78]: if(!isNaN(server)){ (so if the server is a number
floor the number so 5.5 becomes 5 as we dont want to be able acces s5.5 :P

File: commands/adminsonline.js (hunk 9 above)
Comment (tovernaar123) [reply commands/adminsonline.js:81]: well i still check if it exsits but the comment is wrong ye

File: commands/jail.js (hunk 10 above)
Comment (tovernaar123) [reply commands/jail.js:53]: this wont work as the the string with ${} vars has to be with `something here ` not 'some thing here' altough i will change it

File: commands/unjail.js (hunk 14 above)
Comment (tovernaar123) [reply commands/unjail.js:10]: yeah could have been toUnjail but the o next to the u looks wierd :P

File: commands/unjail.js (hunk 17 above)
Comment (tovernaar123) [reply commands/unjail.js:55]: again '' does not allow for ${var} (will change it though)

File: infoBot.js (hunk 21 above)
Comment (tovernaar123) [reply infoBot.js:32]: might make a command to make it so you can disable or enable them

File: rcon_auto_connect.js (hunk 22 above)
Comment (tovernaar123) [reply rcon_auto_connect.js:52]: yep k

File: commands/playersonline-new.js (hunk 4 above)
Comment (tovernaar123) [reply commands/playersonline-new.js]: this file has been deleted lol

**Comment by tovernaar123** [comment]
shood be all fixed

**Pull request merged**